### PR TITLE
Feature/trial events

### PR DESCRIPTION
**Pull request by nwingt**

### CODE REVIEW

**Review by nnkken — APPROVED** [review]

File: src/routes/civic/index.js [hunk 1]
```diff
@@ -133,4 +134,109 @@ router.get('/csonline', async (req, res, next) => {
...
+      .collection('trialEvents')
+      .doc(id)
+      .get();
+
+    if (doc.exists) {
+      const { start, end } = doc.data();
+      const now = Date.now();
+      if (now < start) {
+        res.sendStatus(404);
+      } else if (now > end) {
```
Comment (williamchong) [inline src/routes/civic/index.js]: Use `return` like that api below for easier reading and consistent style?
```
    if (now < start) {
      res.sendStatus(404);
      return;
    }
```

**Review by williamchong — CHANGES_REQUESTED** [review]

File: src/routes/civic/index.js [hunk 2]
```diff
@@ -133,4 +134,113 @@ router.get('/csonline', async (req, res, next) => {
...
+    }
+    if (now > end) {
+      res.sendStatus(410);
+      return;
+    }
+
+    const trialEnd = new Date(now);
+    trialEnd.setMonth(trialEnd.getMonth() + 1);
+    const createObj = {
+      civicLikerStatus: 'subscribed',
```
Comment (mckingho) [inline src/routes/civic/index.js]: suggest the createObj = {
  currentPeriodStart: now,
  currentType: 'trial',
}
A trigger function will update other details.

**Review by mckingho — CHANGES_REQUESTED** [review]

**Comment by mckingho** [comment]
Do we have trial quota?

**Comment by williamchong** [comment]
*Add a `quota` field for each event, which we can set a number higher than the expected users
just in case we got surprised spam

File: src/routes/civic/index.js [hunk 3]
```diff
@@ -213,7 +224,10 @@ router.post('/trial/events/:eventId/join', jwtAuth('write'), async (req, res, ne
       currentPeriodStart: now,
       currentPeriodEnd: trialEnd.getTime(),
     };
-    await subscriptionDbRef.doc(userId).create(createObj);
+    await db.runTransaction(async (t) => {
+      await t.update(trialEventRef, { regCount: regCount + 1 });
```
Comment (williamchong) [inline src/routes/civic/index.js:219]: `regCount` should be `.get()` inside this transaction

**Review by williamchong — CHANGES_REQUESTED** [review]

**Review by williamchong — APPROVED** [review]

**Pull request merged**